.github/workflows/auto-close-github-discussions.yml

```diff
@@ -0,0 +1,304 @@
+name: Auto Close Stale Discussions
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # Runs daily at midnight UTC
+  workflow_dispatch: # Enables manual run from the Actions tab
+
+permissions:
+  contents: read
+  discussions: write
+
+jobs:
+  close-discussions:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Install dependencies
+        run: npm install @octokit/rest @octokit/graphql
+
+      - name: Run inline Discussion Cleanup Script
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          CLOSING_MESSAGE: |
+            Hi,
+
+            As we have not heard back from you, we are closing this discussion to keep our discussions organized.
+
+            Feel free to start a new discussion if this remains relevant.
+
+            Thank you for being part of the community!
+        run: |
+          cat > close-discussions.js <<'EOF'
```

Comment:
Since the script is quite complex, I think it would be better to have it committed as a separate file in the repository in the `scripts` directory. Then the previous step to check out the repository would make sense too.

Reply:
I agree. I have moved the script to a separate file.